fgpyo/fasta/reference_set_builder.py

```diff
@@ -0,0 +1,174 @@
+        """ Place holder for helper funciton """
+        return None
+
+    def to_temp_file(
+        self,
+        delete_on_exit: Optional[bool] = True,
+        calculate_md5_sum: Optional[bool] = False,
+    ) -> None:
+        """
+        For each instance of ReferenceBuilder in REF_BUILDERS write record to temp file
+        """
+        # Write temp file path
+        path = NamedTemporaryFile(
+            prefix=f"{self.assembly}_{self.species}",
+            suffix=".fasta",
+            delete=delete_on_exit,
+            mode=("a+t"),
+        )
+
+        for record in range(len(self.REF_BUILDERS)):
+            seq = self.REF_BUILDERS[record].sequences
+            assembly = self.REF_BUILDERS[record].assembly
+            species = self.REF_BUILDERS[record].species
+            name = self.REF_BUILDERS[record].name
+            header = f">{name}[{assembly}][{species}]\n"
+            seq_format = "\n".join(textwrap.wrap(seq, self.line_length))
+            try:
+                path.write(header)
+                path.write(f"{seq_format}\n\n")
+            except OSError as error:
+                print(f"{error}\nCound not write to {path}")
+
+        # if calculate_md5_sum:
+        # pylint: disable=W0612
+        # with open(path.name, "rb") as path_to_read:
+        # contents = path_to_read.read()
+        # md5 = hashlib.md5(contents).hexdigest()
+
+        # Use md5 to write dict
+        # Write .fai
+
+        path.close()
+
+    def to_file(
+        self,
+        path: str,
+        delete_on_exit: Optional[bool] = True,
+        calculate_md5_sum: Optional[bool] = False,
+    ) -> None:
+        """
+        Same as to_temp_file() but user provides path
+        """
+        with open(path, "a+") as fasta_handle:
+            for record in range(len(self.REF_BUILDERS)):
+                seq = self.REF_BUILDERS[record].sequences
+                assembly = self.REF_BUILDERS[record].assembly
+                species = self.REF_BUILDERS[record].species
+                name = self.REF_BUILDERS[record].name
+                header = f">{name}[{assembly}][{species}]\n"
+                seq_format = "\n".join(textwrap.wrap(seq, self.line_length))
+                try:
+                    fasta_handle.write(header)
+                    fasta_handle.write(f"{seq_format}\n\n")
+                except OSError as error:
+                    print(f"{error}\nCound not write to {path}")
+
+        # if calculate_md5_sum:
+        # with open(path, "rb") as path_to_read:
+        # contents = path_to_read.read()
+        # md5 = hashlib.md5(contents).hexdigest()
+
+        # Use md5 to write dict
+        # Write .fai
+
+        if delete_on_exit:
+            os.remove(path)
+
+
+# pylint: disable=R0903
+class ReferenceBuilder:
+    """
+    Creates individiaul records
+    """
+
+    def __init__(
+        self,
+        name: str,
+        assembly: str,
+        species: str,
+        sequences: Optional[str] = str(),
```

Comment:
Remove this from the constructor, force folks to use the `add` method to add bases
```suggestion
```

Reply:
@nils, I understand the point of forcing people to add bases only through `add` But if `bases` is not an attribute of `ContigBuilder` previously `ReferenceBuilder` do we even need the class? We could just have `FastaBuilder.__contig_builders` be a dict of contig names and bases. Am I misunderstanding something?

Reply:
I think `ContigBuilder` should have an attribute named `bases` that in the constructor is initialized to the empty string, then to add to the bases, the `add` method is used on a `ContigBuilder` instance.  If you look at `FastaBuilder` builder's `add` method, an initial set of bases _is not_ passed (which is what we want), so practically speaking, you have to use the `add` method on `ContigBuilder`.